transaction/core/test-utils/src/lib.rs

```diff
@@ -208,7 +209,9 @@ pub fn create_transaction_with_amount_and_comparer<
     transaction_builder.set_tombstone_block(tombstone_block);
 
     // Build and return the transaction
-    transaction_builder.build_with_sorter::<R, O>(rng).unwrap()
+    transaction_builder
+        .build_with_sorter::<R, O, DummyRingSigner>(&DummyRingSigner {}, rng)
```

Comment:
IIUC we don't need to specify any types here
```suggestion
        .build_with_sorter(&DummyRingSigner {}, rng)
```

Reply:
I think we do, because there is no argument that is bound to the `O` parameter, so rust cannot infer it

Reply:
OK. Can we remove the 3rd type param, since it's passed as an arg?
```suggestion
        .build_with_sorter::<R, O>(&DummyRingSigner {}, rng)
```

Reply:
I think that is not legal in rust, I got an error and that was why I added the type. I don't think you can have some of the type parameters spelled out explicitly but not others.

Reply:
hmm, if the compiler can resolve a type you can often substitute `_`,  so you might be able to `.build_with_sorter::<R, O, _>`?

Reply:
works